public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.test.ts

```diff
@@ -210,3 +299,91 @@ describe('transformSaveModelSchemaV2ToScene', () => {
     expect(getQueryRunnerFor(vizPanels[0])?.state.datasource?.uid).toBe(MIXED_DATASOURCE_NAME);
   });
 });
+
+interface VariableValidation<T extends TypedVariableModelv2> {
```

Comment:
I've abstracted the variable testing. Wanted to get some feedback on this before adding validations for each of these variables.